scala-generator/src/main/scala/models/KafkaConsumer.scala

```diff
@@ -18,6 +19,15 @@ object KafkaConsumer extends CodeGenerator {
     Right(generateCode(form))
   }
 
+  def generateTopicRegex(topicFn: String, apiVersion: String) = {
+    val apiVersionVariable = Seq("${apiVersion}", "$apiVersion").map(Regex.quote(_)).mkString("|")
+    val tenantVariable = Seq("${tenant}", "$tenant").map(Regex.quote(_)).mkString("|")
+    topicFn
+      .replaceAll(apiVersionVariable, Regex.quoteReplacement(apiVersion))
+      //`tenantsPattern` is a val defined in the `topicRegex` function, see `source` below.
```

Comment:
so this function is used to generate the `topicRegex` function:

```scala
def topicRegex(inst: String, tenants: Seq[String]) = {
      val instance = Regex.quote(inst)
      val tenantsPattern = if (tenants.isEmpty) ".*"
                           else tenants.map(Regex.quote(_)).mkString("|")

      ${topicRegex}
    }
```

`tenantsPattern` is a variable defined in that function and what we are doing here is generate that last expression ` ${topicRegex}`, which generates the actual regex at runtime based on the value of `tenants`.